msal/broker.py

```diff
@@ -0,0 +1,202 @@
+    id_token_claims = json.loads(result.get_id_token()) if result.get_id_token() else {}
+    account = result.get_account()
+    assert account, "Account is expected to be always available"
+    ## Note: As of pymsalruntime 0.1.0, only wam_account_ids property is available
+    #account.get_account_property("wam_account_ids")
+    return_value = {k: v for k, v in {
+        "access_token": result.get_access_token(),
+        "expires_in": result.get_access_token_expiry_time(),
+        "id_token_claims": id_token_claims,
+        "client_info": account.get_client_info(),
+        "_account_id": account.get_account_id(),
+        }.items() if v}
+    granted_scopes = result.get_granted_scopes()  # New in pymsalruntime 0.3.x
+    if granted_scopes:
+        return_value["scope"] = " ".join(granted_scopes)  # Mimic the on-the-wire data format
+    return return_value
+
+
+def _get_new_correlation_id():
+    return str(uuid.uuid4())
+
+
+def _signin_silently(authority, client_id, scopes, correlation_id=None, **kwargs):
+    params = pymsalruntime.MSALRuntimeAuthParameters(client_id, authority)
+    params.set_requested_scopes(scopes)
+    callback_data = _CallbackData()
+    for k, v in kwargs.items():  # This can be used to support domain_hint, max_age, etc.
+        if v is not None:
+            params.set_additional_parameter(k, str(v))
+    pymsalruntime.signin_silently(
+        params,
+        correlation_id or _get_new_correlation_id(),
+        lambda result, callback_data=callback_data: callback_data.complete(result))
+    callback_data.signal.wait()
+    return _convert_result(callback_data.auth_result, client_id)
+
+
+def _signin_interactively(
+        authority, client_id, scopes,
+        window=None,
+        prompt=None,
+        login_hint=None,
+        claims=None,
+        correlation_id=None,
+        **kwargs):
+    params = pymsalruntime.MSALRuntimeAuthParameters(client_id, authority)
+    params.set_requested_scopes(scopes)
+    params.set_redirect_uri("placeholder")  # pymsalruntime 0.1 requires non-empty str,
+        # the actual redirect_uri will be overridden by a value hardcoded by the broker
+    if prompt:
+        if prompt == "select_account":
+            if login_hint:
+                # FWIW, AAD's browser interactive flow would honor select_account
+                # and ignore login_hint in such a case.
+                # But pymsalruntime 0.3.x would pop up a meaningless account picker
+                # and then force the account_hint user to re-input password. Not what we want.
+                # https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1744492
+                login_hint = None  # Mimicing the AAD behavior
+                logger.warning("Using both select_account and login_hint is ambiguous. Ignoring login_hint.")
+            params.set_select_account_option(
+                pymsalruntime.SelectAccountOption.SHOWLOCALACCOUNTSCONTROL)
```

Comment:
This will no longer be an available API in the latest release. Lets sync about this sometime. I'll set up a meeting with msalcpp to discuss the remaining prompt types

Reply:
By "latest release", did you mean the already-released 0.5.0? I got an impression that it is coming soon, but not yet in 0.5.0. This PR here currently pins the dependency upper bound as 0.5.x.